src/components/InterestForm/index.tsx

```diff
@@ -0,0 +1,83 @@
+'use client';
+
+import { useState } from 'react';
+import { UUID } from 'crypto';
+import { insertInterest } from '../../api/supabase/queries/interest';
+import { Interest, CaseListing } from '../../types/schema';
+import { Button } from '../Button';
+import {
+  FormContainer,
+  FormTextArea,
+  FormInput,
+  RadioGroup,
+  RadioLabel,
+  RadioInput,
+} from './styles';
+
+const radioOptions = ['Attorney', 'Interpreter', 'Both Attorney & Interpreter'];
+
+export default function InterestForm({ caseData }: { caseData: CaseListing }) {
+  const [reason, setReason] = useState<string>('');
+  const [rolesInterested, setRolesInterested] = useState<string>('');
+  const [startDate, setStartDate] = useState<string>('');
+
+  const handleInsert = async () => {
+    if (reason !== '') {
+      const newInterest: Interest = {
+        // hardcoded values for now
+        id: crypto.randomUUID() as UUID,
+        listing_id: caseData.id,
+        listing_type: 0,
+        user_id: '36b8f84d-df4e-4d49-b662-bcde71a8764f',
+        form_response: {
+          interestType: 'IJP-6-test',
+          interestReason: reason,
+          start_date: startDate,
+        },
+      };
+      await insertInterest(newInterest);
+      setReason('');
+    } else {
+      // console.log('cannot create an empty interest');
+    }
+  };
+
+  return (
+    <FormContainer>
+      {/* <input type="text" /> */}
+      <strong>Submit Interest</strong>
+      <div>What role(s) are you applying for?</div>
+      <RadioGroup>
+        {radioOptions.map(option => (
+          <RadioLabel key={option}>
+            <RadioInput
+              id="radio"
+              type="radio"
+              name="radioOptions"
+              value={option}
+              checked={rolesInterested === option}
```

Comment:
There's a warning about this in the console:

```
Warning: You provided a `checked` prop to a form field without an `onChange` handler. This will render a read-only field. If the field should be mutable use `defaultChecked`. Otherwise, set either `onChange` or `readOnly`.
```

Maybe try to replace `checked` with `defaultChecked` and seeing if that fixes it?

Reply:
changing from onClick to onChange got rid of the console error, thanks for noticing it!